// media-chrome/dist/react/menu/index.js
import React from "react";
import { createComponent } from 'ce-la-react';
import * as Modules from "../../menu/index.js"


export const MediaChromeMenu = createComponent({
  tagName: "media-chrome-menu",
  elementClass: Modules.MediaChromeMenu,
  react: React,
});

export const MediaChromeMenuItem = createComponent({
  tagName: "media-chrome-menu-item",
  elementClass: Modules.MediaChromeMenuItem,
  react: React,
});

export const MediaSettingsMenu = createComponent({
  tagName: "media-settings-menu",
  elementClass: Modules.MediaSettingsMenu,
  react: React,
});

export const MediaSettingsMenuItem = createComponent({
  tagName: "media-settings-menu-item",
  elementClass: Modules.MediaSettingsMenuItem,
  react: React,
});

export const MediaChromeMenuButton = createComponent({
  tagName: "media-chrome-menu-button",
  elementClass: Modules.MediaChromeMenuButton,
  react: React,
});

export const MediaSettingsMenuButton = createComponent({
  tagName: "media-settings-menu-button",
  elementClass: Modules.MediaSettingsMenuButton,
  react: React,
});

export const MediaAudioTrackMenu = createComponent({
  tagName: "media-audio-track-menu",
  elementClass: Modules.MediaAudioTrackMenu,
  react: React,
});

export const MediaAudioTrackMenuButton = createComponent({
  tagName: "media-audio-track-menu-button",
  elementClass: Modules.MediaAudioTrackMenuButton,
  react: React,
});

export const MediaCaptionsMenu = createComponent({
  tagName: "media-captions-menu",
  elementClass: Modules.MediaCaptionsMenu,
  react: React,
});

export const MediaCaptionsMenuButton = createComponent({
  tagName: "media-captions-menu-button",
  elementClass: Modules.MediaCaptionsMenuButton,
  react: React,
});

export const MediaPlaybackRateMenu = createComponent({
  tagName: "media-playback-rate-menu",
  elementClass: Modules.MediaPlaybackRateMenu,
  react: React,
});

export const MediaPlaybackRateMenuButton = createComponent({
  tagName: "media-playback-rate-menu-button",
  elementClass: Modules.MediaPlaybackRateMenuButton,
  react: React,
});

export const MediaRenditionMenu = createComponent({
  tagName: "media-rendition-menu",
  elementClass: Modules.MediaRenditionMenu,
  react: React,
});

export const MediaRenditionMenuButton = createComponent({
  tagName: "media-rendition-menu-button",
  elementClass: Modules.MediaRenditionMenuButton,
  react: React,
});
